shared/chat/inbox/index.desktop.tsx

```diff
@@ -16,57 +16,66 @@ import throttle from 'lodash/throttle'
 import {inboxWidth, getRowHeight} from './row/sizes'
 import {makeRow} from './row'
 import {virtualListMarks} from '../../local-debug'
+import shallowEqual from 'shallowequal'
 
 type State = {
   showFloating: boolean
   showUnread: boolean
 }
 
-class Inbox extends React.PureComponent<T.Props, State> {
+class Inbox extends React.Component<T.Props, State> {
   state = {
     showFloating: false,
     showUnread: false,
   }
 
-  _mounted: boolean = false
-  _list: VariableSizeList | null = null
-  _selectedVisible: boolean = false
+  private mounted: boolean = false
+  private list: VariableSizeList | null = null
+  private selectedVisible: boolean = false
 
   // stuff for UnreadShortcut
-  _firstOffscreenIdx: number = -1
-  _lastVisibleIdx: number = -1
-  _scrollDiv = React.createRef<HTMLDivElement>()
+  private firstOffscreenIdx: number = -1
+  private lastVisibleIdx: number = -1
+  private scrollDiv = React.createRef<HTMLDivElement>()
 
-  componentDidUpdate(prevProps: T.Props) {
+  shouldComponentUpdate(nextProps: T.Props, nextState: State) {
```

Comment:
this changes it so when you expand the inbox it doesn't need to render twice in a row. we instead say we don't need an update, but call the row height invalidating function (which itself forces an update no matter what)